docs/readmes/proposals/p027_magma_outbound_roaming_support.md

```diff
@@ -0,0 +1,136 @@
+
+In this phase, default bearer and general internet access is provided to home routed sessions. Dedicated bearer and other services are out of scope.
+
+##### 3.2.1.4 SubscriberDB (modified)
+
+SubscriberDB will be added to Roaming Gateway to support S6a_proxy originated message handling of AIR, ULR and PUR responses.
+
+##### 3.2.1.5 S8_Proxy (new)
+
+S8_proxy service will implement the GTP-U protocol for establishing sessions between the Visitor SGW and the
+Magma GTP Gateway, the S8-U interface.
+
+##### 3.2.1.6 Control Proxy (modified)
+
+Control Proxy will run on the Roaming Gateway to provide a configuration and message passing interface to the orchestrator. Control proxy and service registry will be updated to support the new GTP_proxy service endpoint.
+
+##### 3.2.1.7 Magmad (modified)
+
+Magmad will run on GTP Gateway to support gateway checkin and configuration management processes. Modifications will be limited to anything needed to handle new mconfig format for new services.
+
+##### 3.2.1.8 Eventd / td-agent-bit (modified)
+
+Eventd and td-agent-bit will run on GTP gateway to support log and metric reporting. Modifications will be to handle new event types and new metrics specific to roaming gateways.
+
+#### 3.2.2 Federation Gateway Services
+
+##### 3.2.2.1 S6a_proxy (modified)
+
+S6a_proxy will be extended on FeG to support AIR, ULR, and PUR messages. These messages will be proxied to the Roaming Gateway where subscriberDB will be running and used to generate response. Content for AIA, ULA, and PUA will be proxied back to FeG where S6a_proxy will translate back to diameter. The main work will be adding support for converting new S6a messages to and from Diameter protocol to gRPC and adding routing to get the gRPC messages from FeG to Roaming Gateway.
+
+New messages to be supported by s6a_proxy:
+
+- Incoming Authentication Information Request (AIR)
+- Incoming User Location Update Request (ULR)
+- Incoming Purge UE Request (PUR)
+
+##### 3.2.2.2 S8_proxy (modified)
+
+S8_proxy service will be extended to support the GTP-C protocol for establishing sessions between the Visitor SGW and the home PGW, the S8-C interface. The northbound interface will add support 3GPP compliant S8-C signaling for the following messages:
+
+- Incoming Create Session Request
+- Incoming Delete Session Request
+- Incoming GTP Echo Request
+
+S8_proxy will be extended to convert the Diameter messages above to gRPC for communication with Roaming Gateway service sessiond.
+
+### 3.3 Call flow
+
+In the call flow below, red number messages indicate new functionality supported by a Magma element.
+
+![call flow](https://user-images.githubusercontent.com/93994458/154710857-f8f23a91-415a-48b9-ab08-cf58450a21c6.png)
```

Comment:
[Mermaid diagrams](https://mermaid-js.github.io/mermaid/) should be preferred over binary blob includes. [They work in Github and DocuSaurus](https://github.com/magma/magma/blob/master/docs/readmes/feg/s1ap_federated_test.md).